src/Response.php

```diff
@@ -328,19 +328,29 @@ public function setContent($content)
 
     /**
      * Sends the response to the client.
+     * @param bool $return Send to stdout or return object. If set to `true` method will return prepared Response object.
+     * @return Response
+     * @throws HeadersAlreadySentException
+     * @throws InvalidConfigException
      */
-    public function send()
+    public function send($return = false)
```

Comment:
Now we have `send()` that does not send. :D

Reply:
How about `getData()`?

Reply:
It's a little confusing. Technically you send it but not in stdout but to the variable.

Reply:
Like with print_r function.

Reply:
Also logically it suppose to call events that you already have in send() when i requesting filled response...

Reply:
Aha. Then maybe `process()` that returns data? Then `send()` will look like `echo process()`.

Reply:
@samdark that can work. What about events `EVENT_BEFORE_SEND` and `EVENT_AFTER_SEND`? Technically we sending content somewhere. Or these events apply only to stdout print?

Reply:
Sending events should be called for sending only. We can add two additional events for processing.  Roadrunner-like server and traditional server are two similar but different models so there may be need to do things a bit differently.